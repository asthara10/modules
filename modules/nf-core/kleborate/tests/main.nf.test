nextflow_process {

    name "Test Process KLEBORATE"
    script "../main.nf"
    process "KLEBORATE"

    tag "modules"
    tag "modules_nfcore"
    tag "kleborate"

    test("test_kleborate") {


        when {
            process {
                """
                
                input[0] = [ 
        [ id:'test', single_end:false ], // meta map
        [ file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true),
          file(params.test_data['sarscov2']['illumina']['contigs_fasta'], checkIfExists: true),
          file(params.test_data['sarscov2']['illumina']['scaffolds_fasta'], checkIfExists: true)
        ]
    ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }

    }

}
