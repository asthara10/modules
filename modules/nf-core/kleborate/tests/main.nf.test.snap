{
    "versions": {
        "content": [
            [
                "versions.yml:md5,d0192c087e9331b9c3a1ce526c21cd1e"
            ]
        ],
        "timestamp": "2023-11-24T13:29:56.530512"
    }
}